boltdb/bucket.go

```diff
@@ -0,0 +1,58 @@
+package boltdb
+
+import (
+	"fmt"
+
+	bolt "go.etcd.io/bbolt"
+)
+
+const (
+	// TopBucket is the bucket at the root level that contains all data related
+	// to projection OCC.
+	TopBucket = "projection_occ"
+)
+
+// bucket retrieves the deepest bucket at a given bucket hierarchy. Slice bb
+// represents the bucket hierarchy with the first element being the top level
+// bucket and the last being the bucket nested at len(bb) level.
+//
+// If tx is writable this function attempts to create any missing buckets in bb.
+// Otherwise, this function produces an error notifying about the first missing
+// bucket encountered in bb.
+func bucket(tx *bolt.Tx, bb ...string) (*bolt.Bucket, error) {
```

Comment:
Could this be two separate functions, one for writing, another for reading, to better express intent?

Reply:
Ok, the latest commits have this function split in two:
1. bucket() for the bucket retrieval within read-only transactions.
1. mkBucketAll() for the bucket retrieval/creation within write transactions.